Upstream: yes
Reason: Fixes a test

From 35ca04ce1b77636f4d929e0d2b83feda081d6684 Mon Sep 17 00:00:00 2001
From: Daniel Stenberg <daniel@haxx.se>
Date: Thu, 18 Feb 2021 08:30:35 +0100
Subject: [PATCH] test1188: change error from connect to resolve error

Using the %NOLISTENPORT to trigger a connection failure is somewhat
"risky" (since it isn't guaranteed to not be listened to) and caused
occasional CI problems. This fix changes the infused error to be a more
reliable one but still verifies the --write-out functionality properly -
which is the purpose of this test.

Reported-by: Jay Satiro
Fixes #6621
Closes #6623
---
 tests/data/test1188 | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/tests/data/test1188 b/tests/data/test1188
index d2e741e4d..8912b391a 100644
--- a/tests/data/test1188
+++ b/tests/data/test1188
@@ -27,7 +27,7 @@ http
 --write-out with %{onerror} and %{urlnum} to stderr
  </name>
 <command>
-http://%HOSTIP:%NOLISTENPORT/we/want/our/1188 http://%HOSTIP:%HTTPPORT/we/want/our/1188 -w '%{onerror}%{stderr}%{urlnum} says %{exitcode} %{errormsg}\n' -s
+http://non-existing-host.haxx.se:%NOLISTENPORT/we/want/our/1188 http://%HOSTIP:%HTTPPORT/we/want/our/1188 -w '%{onerror}%{stderr}%{urlnum} says %{exitcode} %{errormsg}\n' -s
 </command>
 </client>
 
@@ -41,7 +41,7 @@ Accept: */*
 
 </protocol>
 <stderr mode="text">
-0 says 7 Failed to connect to %HOSTIP port %NOLISTENPORT: Connection refused
+0 says 6 Could not resolve host: non-existing-host.haxx.se
 </stderr>
 </verify>
 </testcase>
-- 
2.30.1

